testsuite/Rakefile

```diff
@@ -15,9 +15,14 @@ namespace :cucumber do
       json_results = "-f json -o output_#{timestamp}-#{filename}.json"
       html_results = "-f html -o output_#{timestamp}-#{filename}.html"
       profiles = ENV['PROFILE']
+      # Our profiles include a --tags keyword in all of them, if we have multiple profiles in the list
+      # it will act as an intersection of tags, not as an union of them.
       include_profiles = profiles.nil? ? '--profile default' : '--profile ' + profiles.gsub(',', ' --profile ')
       tags = ENV['TAGS']
-      include_tags =  tags.nil? ? '' : '--tags ' + tags.gsub(',', ' or ')
+      # We provide the list of tags including the '@', so if necessary we can provide a negative tag '~@' on this list.
+      # We generate the tags list as an union of tags, not as an intersection of them (that would be using multiple times `--tags`)
+      # WARNING: When we upgrade Cucumber 2.0.0 to a higher version, we need to use the new Tags expression implementation
+      include_tags =  tags.nil? ? '' : "--tags #{tags}"
```

Comment:
I would prefer passing a list in the non-parallel case (jus a feeling, I am not saying that it won't work):
```
include_tags = tags.nil? ? [] '' : ["--tags", tags]
```
and then pass:
```
cucumber_opts = %W[#{html_results} #{json_results} #{junit_results} -f rerun #{include_profiles} --out failed.txt -f pretty -r features] + include_tags
```

Reply:
I like to keep both tasks with closer code as possible, but you are right, let's use a list.